addons/mrp/models/mrp_production.py

```diff
@@ -1047,3 +1097,51 @@ def _render_note_exception_quantity_mo(rendering_context):
             return self.env.ref('mrp.exception_on_mo').render(values=values)
 
         self.env['stock.picking']._log_activity(_render_note_exception_quantity_mo, documents)
+
+    def _get_planned_duration(self):
+        self.ensure_one()
+        return self.date_planned_finished - self.date_planned_start
+
+    def _dates_depends_on_parent(self):
+        """Return True if the ``mrp.production`` dates might depends on the parent."""
+        self.ensure_one()
+        return self.parent_mrp_production_id and self.state in ['draft', 'confirmed']
```

Comment:
why not 'planned'. It could make sense to reorder the date if the MO is planned but has not start yet.